code/test/util.js

```diff
@@ -215,6 +269,123 @@ const getEOTP = async ({ seed, hseed, effectiveTime, timeOffset }) => {
   return { index, eotp }
 }
 
+// transactionExecute commits and reveals a wallet transaction
+const transactionExecute = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount, data, address, randomSeed, backlinkAddresses, testTime }) => {
+  if (testTime === undefined) { testTime = Date.now() }
+  // // calculate counter from testTime
+  const counter = Math.floor(testTime / INTERVAL)
+  const otp = ONEUtil.genOTP({ seed: wallet.seed, counter })
+  // // calculate wallets effectiveTime (creation time) from t0
+  const info = await wallet.wallet.getInfo()
+  const t0 = new BN(info[3]).toNumber()
+  const walletEffectiveTime = t0 * INTERVAL
+  const index = ONEUtil.timeToIndex({ effectiveTime: walletEffectiveTime, time: testTime })
+  const eotp = await ONE.computeEOTP({ otp, hseed: wallet.hseed })
+
+  // Format commit and revealParams based on tokenType
+  let commitParams
+  let revealParams
+  let paramsHash
+  console.log(`operationType: ${operationType}`)
+  switch (operationType) {
+    case ONEConstants.OperationType.TRACK:
+    case ONEConstants.OperationType.UNTRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
```

Comment:
remove line 293-296? Params are wrong and there should be only one way of handling computeGeneralOperationHash

Reply:
Currently when removing these lines and adding the additional `amount` and `data` fields` and run
`VERBOSE=0 truffle test --network=ganache --compile-none --grep 'OPERATION 0 '`
with the following parameters
```
   testTime = await TestUtil.bumpTestTime(testTime, 60)
    await TestUtil.executeStandardTransaction(
      {
        ...ONEConstants.NullOperationParams, // Default all fields to Null values than override
        wallet: alice,
        operationType: ONEConstants.OperationType.TRACK,
        tokenType: ONEConstants.TokenType.ERC20,
        contractAddress: testerc20.address,
        testTime
      }
    )
```
      
we get
```
     Error: Returned error: VM Exception while processing transaction: revert Param mismatch
      at commitReveal (test/util.js:417:44)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at Object.executeStandardTransaction (test/util.js:378:3)
      at Context.<anonymous> (test/general.js:53:5)
 ```
This appears to be because of the logic in Executor.sol testing the op.data.length
```
        // No revert should occur below this point
        if (op.operationType == Enums.OperationType.TRACK) {
            if (op.data.length > 0) {
                tokenTrackerState.multiTrack(op.data);
            } else {
                tokenTrackerState.trackToken(op.tokenType, op.contractAddress, op.tokenId);
            }
        } else if (op.operationType == Enums.OperationType.UNTRACK) {
            if (op.data.length > 0) {
                tokenTrackerState.multiUntrack(op.data);
            } else {
                tokenTrackerState.untrackToken(op.tokenType, op.contractAddress, op.tokenId);
            }
```
Hence if we pass data we (incorrectly) call `multitrack` if we omit data by using say the following parameters
```
       //...ONEConstants.NullOperationParams, // Default all fields to Null values than override
        wallet: alice,
        operationType: ONEConstants.OperationType.TRACK,
        tokenType: ONEConstants.TokenType.ERC20,
        contractAddress: testerc20.address,
        TokenId: 1,
        dest: alice.wallet.address,
        amount: 1,
        testTime
        ```
        We get
        ```
             Error: invalid BigNumber value (argument="value", value=undefined, code=INVALID_ARGUMENT, version=bignumber/5.4.1)
      at commitReveal (test/util.js:417:44)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at Object.executeStandardTransaction (test/util.js:378:3)
      at Context.<anonymous> (test/general.js:53:5)
      ```
      Hence the workaround for now.

Reply:
Client side implementations (stuff in client/src/pages/Show/*.jsx) all use separate commit params (and hash computing function) and reveal params (and reveal args transformation function, if needed). Here you are attempting to make one function that encompass both commit and reveal functionalities, which have incompatible parameter names, hence the errors / difficulties